File: examples/CBuildr3/ExNbHntU.cpp
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include "ExNbHntU.h"
//---------------------------------------------------------------------------
#pragma link "OvcNbk"
#pragma link "OvcBase"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::NB1MouseOverTab(TObject *Sender, int Index)
{
  CurIndex = Index;

  // If over a tab, set global flag to true
  OverTab = true;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::FormCreate(TObject *Sender)
{
  Application->OnShowHint = DoShowHint;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::DoShowHint(String& HintStr, bool& CanShow, THintInfo& HintInfo)
{
  TRect R;
  TPoint P;
  int VOffSet;

  if (HintInfo.HintControl == NB1) {
    R = NB1->GetTabRect(CurIndex);
    HintInfo.CursorRect = R;
    P.x = R.Left;
    P.y = R.Top;

    P = ClientToScreen(P);

    HintInfo.HintPos.x = P.x + 20;
    VOffSet = (R.Bottom - R.Top) * (3/2);
    HintInfo.HintPos.y = P.y + VOffSet;

    HintStr = NB1->Pages[CurIndex]->Hint;
  }
  // Show hint only if over a tab
  CanShow = OverTab;

  // Force global flag false so hint is not shown unless over a tab
  OverTab = false;
}

File: examples/Delphi/EXNBHNTU.pas
{$J+} {Writable constants}

unit Exnbhntu;

interface

uses
  SysUtils, WinTypes, WinProcs, Messages, Classes, Graphics, Controls,
  Forms, Dialogs, OvcBase, OvcNbk, StdCtrls;

type
  TForm1 = class(TForm)
    NB1: TOvcNotebook;
    OvcController1: TOvcController;
    Memo1: TMemo;
    procedure NB1MouseOverTab(Sender: TObject; Index: Integer);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
    CurIndex : Integer;
    OverTab  : Boolean;

    procedure DoShowHint(var HintStr: string; var CanShow: Boolean;
                         var HintInfo: THintInfo);
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

procedure TForm1.NB1MouseOverTab(Sender: TObject; Index: Integer);
begin
  CurIndex := Index;

  {If over a tab, set global flag to True}
  OverTab := True;
end;

procedure TForm1.FormCreate(Sender: TObject);
begin
  Application.OnShowHint := DoShowHint;
end;


procedure TForm1.DoShowHint(var HintStr: string; var CanShow: Boolean;
   var HintInfo: THintInfo);
var
  R : TRect;
  P : TPoint;
  VOffSet : Integer;
begin
  with HintInfo do begin
    if HintControl = NB1 then begin
      R := NB1.GetTabRect(CurIndex);
      CursorRect := R;
      P.X := R.Left;
      P.Y := R.Top;

      P := ClientToScreen(P);

      HintPos.X := P.X + 20;
      VOffSet := (R.Bottom - R.Top) * 3 div 2;
      HintPos.Y := P.Y + VOffSet;

      HintStr := NB1.Pages[CurIndex].Hint;
    end;
  end;

  {Show hint only if over a tab}
  CanShow := OverTab;

  {Force global flag False so hint is not shown unless over a tab}
  OverTab := False;
end;


end.

